### ISSUE TITLE
Mempool should prioritize txs based on StandardTx().Fee value

### ISSUE THREAD

**Issue by ggoranov** [dusk-network/dusk-blockchain#169]

**Comment by ggoranov** [comment]
Using sort.Slice at each pool.Put call performs much slower than with disabled sorting.
Results below:

Disabled `sort.Slice` sorting
``` 
BenchmarkPut-8                 1        5344484857 ns/op
--- BENCH: BenchmarkPut-8
    hashmap_test.go:126: Pool transactions count: 30000
PASS
ok      github.com/dusk-network/dusk-blockchain/pkg/core/mempool        5.417s

```

Enabled  `sort.Slice` sorting

```
--- BENCH: BenchmarkPut-8
    hashmap_test.go:126: Pool transactions count: 30000
PASS
ok      github.com/dusk-network/dusk-blockchain/pkg/core/mempool        50.749s

```

BinarySearch  and data shift without sort.Slice

```
BenchmarkPut-8                 1        6305648889 ns/op
--- BENCH: BenchmarkPut-8
    hashmap_test.go:126: Pool transactions count: 30000
PASS
ok      github.com/dusk-network/dusk-blockchain/pkg/core/mempool        6.379s

```

**Comment by autholykos** [comment]
Closed via #171
